res/controllers/Midi_for_light-scripts.js

```diff
@@ -107,14 +111,61 @@ midi_for_light.shutdown = function(id) { // called when the MIDI device is close
             engine.stopTimer(deck_beat_watchdog_timer[i]);
         }
     }
-    for (const timer of ["vu_meter_timer", "volumeBeatBlock_timer", "crossfader_change_block_timer", "volumebeat_on_delay_timer"]) {
+    for (const timer of ["vu_meter_timer"]) {
         if (midi_for_light[timer]) {
             engine.stopTimer(midi_for_light[timer]);
             midi_for_light[timer] = undefined;
         }
     }
 };
 
+midi_for_light.calculateDeckPriority = function() {
+    // Calculate each channels Volume to figure out the most important
+    var crossfader = engine.getValue("[Master]", "crossfader");
+    var crossfader_left =  Math.min((1-crossfader)*1.33,1);
+    var crossfader_right =  Math.min((1+crossfader)*1.33,1);
+    var crossfader_factors = [crossfader_left, 1.0, crossfader_right]
+    
+    for (var i = 0; i < 4; i++) {
+        var channel = "[Channel" + (i + 1) + "]";
+        midi_for_light.decks[i].playing = engine.getParameter(channel, "play") == 1;
+        if (! midi_for_light.decks[i].playing) {
+            midi_for_light.decks[i].priority = 0.0;
+            continue;
+        }
+
+        midi_for_light.decks[i].priority = engine.getParameter(channel, "volume") * crossfader_factors[engine.getValue(channel, "orientation")];
+
+        // Decrease Priority of ending Tracks
+        var duration = engine.getValue(channel, "duration");
+        var PlayPosition = duration * engine.getValue(channel, "playposition");
+        if (duration - PlayPosition < deck_ending_time) {
```

Comment:
```suggestion
        var playPosition = duration * engine.getValue(channel, "playposition");
        if (duration - playPosition < deck_ending_time) {
```

for consistency. I would, by the way, also recommend using `const` or `let` in new code where possible, even if other parts of the script still use `var`.